mobile/lib/ui/viewer/actions/album_selection_action_widget.dart

```diff
@@ -0,0 +1,372 @@
+
+      items.add(
+        SelectionActionButton(
+          labelText: "Unpin",
+          icon: CupertinoIcons.pin_slash,
+          onTap: _onUnpinClick,
+          shouldShow: hasPinnedAlbum,
+        ),
+      );
+
+      items.add(
+        SelectionActionButton(
+          labelText: S.of(context).delete,
+          icon: Icons.delete_outline,
+          onTap: _trashCollection,
+        ),
+      );
+      items.add(
+        SelectionActionButton(
+          labelText: S.of(context).hide,
+          icon: Icons.visibility_off_outlined,
+          onTap: _onHideClick,
+        ),
+      );
+    }
+
+    items.add(
+      SelectionActionButton(
+        labelText: S.of(context).archive,
+        icon: Icons.archive_outlined,
+        onTap: _archiveClick,
+      ),
+    );
+
+    if (widget.sectionType == UISectionType.incomingCollections) {
+      items.add(
+        SelectionActionButton(
+          labelText: S.of(context).leaveAlbum,
+          icon: Icons.logout,
+          onTap: _leaveAlbum,
+        ),
+      );
+    }
+
+    final scrollController = ScrollController();
+
+    return MediaQuery(
+      data: MediaQuery.of(context).removePadding(removeBottom: true),
+      child: SafeArea(
+        child: Scrollbar(
+          radius: const Radius.circular(1),
+          thickness: 2,
+          controller: scrollController,
+          thumbVisibility: true,
+          child: SingleChildScrollView(
+            physics: const BouncingScrollPhysics(
+              decelerationRate: ScrollDecelerationRate.fast,
+            ),
+            scrollDirection: Axis.horizontal,
+            child: Container(
+              padding: const EdgeInsets.only(bottom: 24),
+              child: Row(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: [
+                  const SizedBox(width: 4),
+                  ...items,
+                  const SizedBox(width: 4),
+                ],
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  Future<void> _shareCollection() async {
+    await routeToPage(
+      context,
+      AddParticipantPage(
+        widget.selectedAlbums.albums.toList().first,
+        false,
+        collections: widget.selectedAlbums.albums.toList(),
+      ),
+    );
+    if (hasFavorites) {
+      _showFavToast();
+    }
+    widget.selectedAlbums.clearAll();
+  }
+
+  Future<void> _trashCollection() async {
+    int count = 0;
+    final List<Collection> nonEmptyCollection = [];
+
+    for (final collection in widget.selectedAlbums.albums) {
+      count = await FilesDB.instance.collectionFileCount(collection.id);
+      final bool isEmptyCollection = count == 0;
+      if (isEmptyCollection) {
+        try {
+          await CollectionsService.instance.trashEmptyCollection(collection);
+        } catch (e, s) {
+          _logger.warning("failed to trash collection", e, s);
+          await showGenericErrorDialog(context: context, error: e);
```

Comment:
If trashing fails for multiple empty collections, this will result in multiple back to back dialogs.